http-service-lambda/src/lib.rs

```diff
@@ -0,0 +1,179 @@
+//! `HttpService` server that uses AWS Lambda Rust Runtime as backend.
+//!
+//! This crate builds on the standard http interface provided by the
+//! [lambda_http](https://docs.rs/lambda_http) crate and provides a http server
+//! that runs on the lambda runtime.
+//!
+//! Compatible services like [tide](https://github.com/rustasync/tide) apps can
+//! run on lambda and processing events from API Gateway or ALB without much
+//! change.
+//!
+//! # Examples
+//!
+//! **Hello World**
+//!
+//! ```rust,ignore
+//! #![feature(async_await)]
+//!
+//! fn main() {
+//!     let mut app = tide::App::new();
+//!     app.at("/").get(async move |_| "Hello, world!");
+//!     http_service_lambda::run(app.into_http_service());
+//! }
+//! ```
+
+#![forbid(future_incompatible, rust_2018_idioms)]
+#![deny(missing_debug_implementations, nonstandard_style)]
+#![warn(missing_docs, missing_doc_code_examples)]
+#![cfg_attr(test, deny(warnings))]
+#![feature(async_await)]
+
+use futures::{FutureExt, TryFutureExt};
+use http_service::{Body as HttpBody, HttpService, Request as HttpRequest};
+use lambda_http::{Body as LambdaBody, Handler, Request as LambdaHttpRequest};
+use lambda_runtime::{error::HandlerError, Context};
+use std::future::Future;
+use std::sync::Arc;
+use tokio::runtime::Runtime as TokioRuntime;
+
+type LambdaResponse = lambda_http::Response<lambda_http::Body>;
+
+trait ResultExt<OK, ERR> {
+    fn handler_error(self, description: &str) -> Result<OK, HandlerError>;
+}
+
+impl<OK, ERR> ResultExt<OK, ERR> for Result<OK, ERR> {
+    fn handler_error(self, description: &str) -> Result<OK, HandlerError> {
+        self.map_err(|_| HandlerError::from(description))
+    }
+}
+
+trait CompatHttpBodayAsLambda {
```

Comment:
Not a big deal since it's a private trait, but spelling mistake in this trait name

```suggestion
trait CompatHttpBodyAsLambda {
```

Reply:
Good catch!